lab_1_keywords_tfidf/main.py

```diff
@@ -48,7 +65,17 @@ def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(tokens, list):
+        return None
+
+    frequencies = {}
+    for i in tokens:
+        if not isinstance(i, str):
+            return None
+        freq = tokens.count(i)
```

Comment:
Why do you need `freq`? Can you set `tokens.count(i)` directly to `frequencies[i]`?

Reply:
fixed